### DISCUSSION TITLE
How to determine if jemalloc is being used

### DISCUSSION THREAD

**Discussion by driv** [fluent/fluent-bit#9529] [Q&A] [answered]
I'm looking into an instance of fluent-bit running in a container.

I would like to confirm that jemalloc is being used, but by running the container I've not been able to confirm it. From the documentation I understand that the production docker image should be using it but I wanted to be sure.

I've tried using pmap or lsof, but I don't see the library being referenced.

**Comment by edsiper** [comment]
You can use the `-h` option and check the built-in flags, note that this can only be done with the `-debug` image:

```bash
docker run -ti fluent/fluent-bit:3.1.9-debug /fluent-bit/bin/fluent-bit -h |grep FLB_HAVE_JEMALLOC
 Build Flags = FLB_HAVE_SYS_WAIT_H FLB_HAVE_IN_STORAGE_BACKLOG FLB_HAVE_CHUNK_TRACE FLB_HAVE_PARSER
 FLB_HAVE_RECORD_ACCESSOR FLB_HAVE_STREAM_PROCESSOR NGHTTP2_STATICLIB JSMN_PARENT_LINKS JSMN_STRICT 
FLB_EVENT_LOOP_AUTO_DISCOVERY FLB_HAVE_TLS FLB_HAVE_OPENSSL FLB_HAVE_METRICS FLB_HAVE_WASM 
FLB_HAVE_AWS FLB_HAVE_AWS_CREDENTIAL_PROCESS FLB_HAVE_SIGNV4 FLB_HAVE_SQLDB FLB_LOG_NO_CONTROL_CHARS 
FLB_HAVE_METRICS FLB_HAVE_HTTP_SERVER FLB_HAVE_SYSTEMD FLB_HAVE_SYSTEMD_SDBUS FLB_HAVE_FORK 
FLB_HAVE_TIMESPEC_GET FLB_HAVE_GMTOFF FLB_HAVE_UNIX_SOCKET FLB_HAVE_LITTLE_ENDIAN_SYSTEM 
FLB_HAVE_LIBYAML FLB_HAVE_ATTRIBUTE_ALLOC_SIZE FLB_HAVE_PROXY_GO FLB_HAVE_JEMALLOC JEMALLOC_MANGLE 
FLB_HAVE_LIBBACKTRACE FLB_HAVE_REGEX FLB_HAVE_UTF8_ENCODER FLB_HAVE_LUAJIT FLB_HAVE_C_TLS 
FLB_HAVE_ACCEPT4 FLB_HAVE_INOTIFY FLB_HAVE_GETENTROPY FLB_HAVE_GETENTROPY_SYS_RANDOM
 ```
 
 
![image](https://github.com/user-attachments/assets/3ec7ccf9-fb59-4aee-a329-da0e7f5656f9)

**Reply by driv** [reply]
Yes, the flag is there, but I don't see the library being loaded with `pmap`, that is making me wonder if anything else needs to be done to make fluent-bit use it.

BTW, it also works in the production image:
```
docker run --rm --name fluent-bit cr.fluentbit.io/fluent/fluent-bit:latest -h
```

**Reply by patrick-stephens** [reply, marked as answer]
I thought it was statically linked in but may be wrong.
